packages/components/src/navigation/README.md

```diff
@@ -0,0 +1,192 @@
+# Navigation
+
+Render a navigation list with optional groupings and hierarchy.
+
+## Usage
+
+```jsx
+import {
+	__experimentalNavigation as Navigation,
+	__experimentalNavigationGroup as NavigationGroup,
+	__experimentalNavigationItem as NavigationItem,
+	__experimentalNavigationLevel as NavigationLevel,
+} from '@wordpress/components';
+
+const MyNavigation = () => (
+	<Navigation>
+		<NavigationLevel title="Home">
+			<NavigationGroup title="Group 1">
+				<NavigationItem item="item-1" title="Item 1" />
+				<NavigationItem item="item-2" title="Item 2" />
+			</NavigationGroup>
+			<NavigationGroup title="Group 2">
+				<NavigationItem
+					item="item-3"
+					navigateToLevel="category"
+					title="Category"
+				/>
+			</NavigationGroup>
+		</NavigationLevel>
+
+		<NavigationLevel
+			level="category"
+			parentLevel="root"
+			parentLevelTitle="Home"
+			title="Category"
+		>
+			<ul>
+				<NavigationItem
+					badge="1"
+					item="child-1"
+					title="Child 1"
+				/>
+				<NavigationItem item="child-2" title="Child 2" />
+			</ul>
+		</NavigationLevel>
+	</Navigation>
+);
+```
+
+## Navigation Props
+
+`Navigation` supports the following props.
+
+### `activeItem`
+
+-   Type: `string`
+-   Required: No
+
+The active item slug.
+
+### `activeLevel`
+
+-   Type: `string`
+-   Required: No
+-   Default: "root"
+
+The active level slug.
+
+### className
+
+-   Type: `string`
+-   Required: No
+
+Optional className for the `Navigation` component.
+
+### `setActiveItem`
+
+-   Type: `function`
+-   Required: No
+
+Sync the active item between the external state and the Navigation's internal state.
+
+### `setActiveLevel`
+
+-   Type: `function`
+-   Required: No
+
+Sync the active level between the external state and the Navigation's internal state.
+
+## Navigation Level
+
+`NavigationLevel` supports the following props.
+
+### className
+
+-   Type: `string`
+-   Required: No
+
+Optional className for the `NavigationLevel` component.
+
+### `level`
+
+-   Type: `string`
+-   Required: No
+-   Default: "root"
+
+The level slug.
+
+### `parentLevel`
+
+-   Type: `string`
+-   Required: No
+
+The parent level slug; used by nested levels to indicate their parent level.
+
+### `parentLevelTitle`
```

Comment:
Yup, the `parentLevelTitle` is unnecessary, and it might make more sense to call it `backButtonLabel` instead, since technically that's what it does.

Inferring it from the `Navigation` children is doable, but it would require a big enough change + possible discussion on the data handling, so I've delayed to a follow up. 🙂